packages/truffle-deployer/index.js

```diff
@@ -42,8 +43,14 @@ class Deployer extends Deployment {
   deploy() {
     const args = Array.prototype.slice.call(arguments);
     const contract = args.shift();
+    const networkType = this.networks[this.network].type;
 
-    if (this.networks[this.network].type === "quorum") {
+    if (
+      networkType &&
+      getSupportedNetworks().then(supportedNetworks =>
+        supportedNetworks.includes(networkType)
+      )
```

Comment:
This syntax is wrong: the `getSupportedNetworks().then(/*...*/)` bit will _always_ resolve to a truthy value

Reply:
Best make `getSupportedNetworks` synchronous, I think